src/norm/pool.nim

```diff
@@ -0,0 +1,99 @@
+import std/locks
+
+import sqlite, postgres
+
+
+type 
+  DbConn = sqlite.DbConn | postgres.DbConn
+  Pool*[T: DbConn] = ref object 
+    defaultSize: Natural
+    conns: seq[T]
+    poolExhaustedPolicy: PoolExhaustedPolicy
+    getDbProc: proc: T
+    lock: Lock
+  PoolExhaustedError* = object of CatchableError
+  PoolExhaustedPolicy* = enum
+    pepRaise
+    pepExtend
+
+
+proc newPool*[T: DbConn](defaultSize: Positive, poolExhaustedPolicy = pepRaise, getDbProc: proc: T = getDb): Pool[T] =
```

Comment:
Does that actually get past the tests?
I'm a bit surprised, I'd have assumed that the compiler starts nagging about how it can't figure out whether to use sqlite.DbConn or postgres.DbConn

Reply:
Given that the tests ran through, it actually does work... I'm so confused. Oh well, nice to see it function!